src/app/clusters/laundry-washer-controls-server/laundry-washer-controls-server.cpp

```diff
@@ -0,0 +1,199 @@
+
+private:
+    CHIP_ERROR ReadSpinSpeeds(AttributeValueEncoder & aEncoder, Delegate * delegate);
+    CHIP_ERROR ReadSpinSpeedCurrent(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadNumberOfRinses(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadSupportedRinses(AttributeValueEncoder & aEncoder, Delegate * delegate);
+};
+
+CHIP_ERROR LaundryWasherControlsAttrAccess::ReadSpinSpeeds(AttributeValueEncoder & aEncoder, Delegate * delegate)
+{
+    return delegate->HandleGetSpinSpeedsList(aEncoder);
+}
+
+CHIP_ERROR LaundryWasherControlsAttrAccess::ReadSpinSpeedCurrent(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    DataModel::Nullable<uint8_t> speedSetting;
+    SpinSpeedCurrent::Get(endpoint, speedSetting);
+    uint8_t ret = 0;
+    if (!speedSetting.IsNull())
+    {
+        ret = speedSetting.Value();
+    }
+
+    return aEncoder.Encode(ret);
+}
+
+CHIP_ERROR LaundryWasherControlsAttrAccess::ReadNumberOfRinses(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    LaundryWasherControls::NumberOfRinsesEnum numOfRinses;
+    NumberOfRinses::Get(endpoint, &numOfRinses);
+
+    return aEncoder.Encode(numOfRinses);
+}
+
+CHIP_ERROR LaundryWasherControlsAttrAccess::ReadSupportedRinses(AttributeValueEncoder & aEncoder, Delegate * delegate)
+{
+    return delegate->HandleGetSupportedRinses(aEncoder);
+}
+
+CHIP_ERROR LaundryWasherControlsAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    EndpointId endpoint = aPath.mEndpointId;
+    Delegate * delegate = GetDelegate(endpoint);
+
+    if (aPath.mClusterId != LaundryWasherControls::Id)
+    {
+        // We shouldn't have been called at all.
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    switch (aPath.mAttributeId)
+    {
+    case Attributes::SpinSpeeds::Id:
+        if (isDelegateNull(delegate, endpoint))
+        {
+            return aEncoder.EncodeEmptyList();
+        }
+        return ReadSpinSpeeds(aEncoder, delegate);
+    case Attributes::SpinSpeedCurrent::Id:
+        return ReadSpinSpeedCurrent(endpoint, aEncoder);
+    case Attributes::NumberOfRinses::Id:
+        return ReadNumberOfRinses(endpoint, aEncoder);
+    case Attributes::SupportedRinses::Id:
+        if (isDelegateNull(delegate, endpoint))
+        {
+            return aEncoder.EncodeEmptyList();
+        }
+        return ReadSupportedRinses(aEncoder, delegate);
+    default:
+        break;
+    }
+    return CHIP_NO_ERROR;
+}
+} // namespace
+
+void emberAfLaundryWasherControlsClusterInitCallback(chip::EndpointId endpoint)
+{
+    return;
+}
+
+using imcode = Protocols::InteractionModel::Status;
+
+void MatterLaundryWasherControlsClusterServerAttributeChangedCallback(const chip::app::ConcreteAttributePath & attributePath)
+{
+    // ToDo
+    return;
+}
+
+Status MatterLaundryWasherControlsClusterServerPreAttributeChangedCallback(const chip::app::ConcreteAttributePath & attributePath,
```

Comment:
Why is this here at all?

I expect this is not called anyway, because the cluster is not annotated to call it, right?

Reply:
I will remove them